### PR TITLE
add support for `default_version` in control file

### LINKED ISSUE #147: Use default_version information from control file
Currently the `default_version` from control files is not tracked

We recommend that users always specify a version number when enabling an extension but this argument should be added to the top level package table and updated on `dbdev publish` because it has an impact on pg_dump/pg_restore projects.

Note, there are 3 components to this:
- tracking `default_version` in the database
- (if needed) create a new release of the dbdev in-database client that respects the value
- (if needed) update the dbdev CLI to respect the value

### PR DESCRIPTION
**Pull request by imor**
## What kind of change does this PR introduce?

feature

## What is the current behavior?

`default_version` key in the control file is not sent to the backend when a user runs `dbdev publish`. This results in TLE's default version being the first installed version which is usually the oldest version.

## What is the new behavior?

The `default_version` is now tracked in the `app.packages` table in the `default_version_struct` and `default_version` columns. The `dbdev` CLI also sends the `default_version` from the control file to the backend when running `dbdev publish` command. The CLI also checks that the `default_version` in the control file is one of the valid versions of the package. The in-db client has been updated to fetch the `default_version` from the `public.packages` view and call the `pgtle.set_default_version` function.

Because old dbdev CLI versions are not aware of `default_version`, the `default_version_struct` and `default_version` columns in `app.packages` are nullable for backward compatibility with those older CLI versions.

## Additional context

Fixes #147

### CODE REVIEW

**Review by olirice — CHANGES_REQUESTED** [review]
> Because old dbdev CLI versions are not aware of default_version, the default_version_struct and default_version columns in app.packages are nullable for backward compatibility with those older CLI versions.

The current user base is like 5 people and we're pre-1.0. I think it's worth taking a breaking change thats solvable with a CLI upgrade to avoid having to make the field nullable at this point. wdyt?

**Comment by imor** [comment]
> > Because old dbdev CLI versions are not aware of default_version, the default_version_struct and default_version columns in app.packages are nullable for backward compatibility with those older CLI versions.
> 
> The current user base is like 5 people and we're pre-1.0. I think it's worth taking a breaking change thats solvable with a CLI upgrade to avoid having to make the field nullable at this point. wdyt?

`default_version` is important to be set so making it non-nullable definitely reduces chances of bugs with the added benefit of making the code in this PR a lot simpler. So let me make this change.

File: supabase/migrations/20231205051816_add_default_version.sql [hunk 1]
```diff
@@ -0,0 +1,95 @@
+-- default_version column has a default value '0.0.0' only temporarily because the column is not null.
+-- It will be removed below.
+alter table app.packages
+add column default_version_struct app.semver not null default app.text_to_semver('0.0.0'),
```
Comment (imor) [inline supabase/migrations/20231205051816_add_default_version.sql:4]: `default_version_struct` is not longer nullable.

File: supabase/migrations/20231205051816_add_default_version.sql [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    account app.accounts = account from app.accounts account where id = auth.uid();
+    require text;
+begin
+    if account.handle is null then
+        raise exception 'user not logged in';
+    end if;
+
+    if default_version is null then
+        raise exception 'default_version is required. If you are on `dbdev` CLI version 0.1.5 or older upgrade to the latest version.';
+    end if;
```
Comment (imor) [inline supabase/migrations/20231205051816_add_default_version.sql:72]: If the client doesn't send `default_version` we hint in the error message for older clients to be upgraded to the latest.

**Pull request merged**